scipy/_lib/_array_api.py

```diff
@@ -711,21 +710,22 @@ def make_skip_xp_backends(*funs, capabilities_table=None):
 
     skip_backends = []
     cpu_only = False
-    cpu_only_reason = ""
+    cpu_only_reason = set()
     np_only = False
     exceptions = []
 
     for fun in funs:
         skip_backends += capabilities_table[fun].get('skip_backends', [])
         cpu_only |= capabilities_table[fun].get('cpu_only', False)
-        cpu_only_reason += capabilities_table[fun].get('reason', "")
+        # Empty reason causes the decorator to have no effect
+        cpu_only_reason.add(capabilities_table[fun].get('reason', "No reason given."))
```

Comment:
For whatever reason, if an empty `reason` is passed as a keyword argument, the `cpu_only` decorator doesn't do its job. Simplest fix to get CI passing (hopefully) is to provide a reason. If this is all going away, that should be find to get the rest of this PR merged.